### ISSUE TITLE
ClassOrdering Rule produces a misleading error message

### ISSUE THREAD

**Issue by n0rthdev** [detekt/detekt#7456]
The following code produces the expected behavior

`secondary constructor should be declared after property declarations and initializer blocks. [ClassOrdering]`

```kotlin
class MyClass(
    var a: String
) {
    constructor() : this("")

    val prop1: Boolean
        get() = false

    val prop2: Boolean
        get() = true
}
```

However, when adding a function at the bottom, the Behavior changes to

`secondary constructor should be declared before method declarations. [ClassOrdering]`

```kotlin
class MyClass(
    var a: String
) {
    constructor() : this("")

    val prop1: Boolean
        get() = false

    val prop2: Boolean
        get() = true

    fun fun1() = true
}
```

If I now remove the prop2. I again get a valid error message: 

`property `prop1` should be declared before secondary constructors. [ClassOrdering]`

```kotlin
class MyClass(
    var a: String
) {
    constructor() : this("")

    val prop1: Boolean
        get() = false

    fun fun1() = true
}
```

## Expected Behavior
`secondary constructor should be declared after property declarations and initializer blocks. [ClassOrdering]`

## Observed Behavior
`secondary constructor should be declared before method declarations. [ClassOrdering]`

## Your Environment

* Version of detekt used: 1.23.6  (with kotlin version 1.9.24)
* Version of Gradle used (if applicable): 8.8
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version: Windows 11
* Link to your project (if it's a public repository):

**Comment by cortinico** [comment]
Yup this looks like a bug in the rule logic 👍

**Comment by n0rthdev** [comment]
> s like a bug in the r

I would like to take a look at it and try to fix it, if this is fine for you?

**Comment by n0rthdev** [comment]
The current main version (https://github.com/detekt/detekt/commit/4fa535bf0fe891dabf7ffef420a408130d91fce2) seems to have this issue fixed already. 

```
class MyClass(
    var a: String
) {
    constructor() : this("")

    val prop1: Boolean
        get() = false

    val prop2: Boolean
        get() = true

    fun fun1() = true
}
```

is now reporting:
`secondary constructor should be declared after property declarations and initializer blocks.`

**Comment by 3flex** [comment]
Probably fixed by https://github.com/detekt/detekt/pull/7423

If the behaviour is now correct this can be closed.

**Comment by n0rthdev** [comment]
@3flex yes, I would consider this fixed.
